index.js

```diff
@@ -18,19 +18,28 @@ function getEnvironment (options) {
 exports.name = 'nunjucks';
 exports.outputFormat = 'html';
 
-exports.render = function (str, options, locals) {
+exports.compile = function (str, options) {
   var env = getEnvironment(options)
-  return env.renderString(str, locals)
+  var tmpl = new Template(str, env, true)
+  return function (locals) {
+    return tmpl.render(locals)
+  }
```

Comment:
It expected `this` to be set. Returning `tmpl.render` won't do that unfortunately.